test_pull_requests

```diff
@@ -451,34 +570,27 @@ module Hub
     def mark_test_success(pull_id, repo, settings)
       $stderr.puts "  Marking SUCCESS for pull request ##{pull_id} in repo '#{repo}'"
       bot_comment = get_comment_with_prefix(pull_id, repo, settings['test_prefix'])
-      build_url = nil
-      extended_tests = nil
-      if bot_comment['body'] =~ /^#{settings['test_prefix']} Running \(([^\)]+)\)( \(Extended Tests: [^\)]+\).*)?/
-        build_url = $1
-        extended_tests = $2
-      else
-        raise "'#{settings['name']}' pull request not found"
-      end
-      comment = "#{settings['test_prefix']} SUCCESS (#{build_url})"
-      comment += extended_tests if extended_tests
+      fields = extract_bot_comment_fields(bot_comment['body'], settings)
+      fields[:state] = :success
+      comment = compose_bot_comment(settings['test_prefix'], fields)
       recreate_comment(pull_id, bot_comment['id'], repo, comment)
 
       commit = get_last_commit(pull_id, repo)
-      update_status(settings['test_prefix'], commit['sha'], repo, 'success', build_url, 'Passed')
+      update_status(settings['test_prefix'], commit['sha'], repo, 'success', fields[:build_url], 'Passed')
```

Comment:
This looks a little suspect -- maybe we should refactor the status update to take your state hash as well